src/Plugin/Field/FieldWidget/ModeNumberWidget.php

```diff
@@ -0,0 +1,248 @@
+<?php
+
+namespace Drupal\commerce\Plugin\Field\FieldWidget;
+
+use Drupal\Core\Field\FieldFilteredMarkup;
+use Drupal\Core\Field\FieldItemListInterface;
+use Drupal\Core\Field\WidgetBase;
+use Drupal\Core\Form\FormStateInterface;
+use Symfony\Component\Validator\ConstraintViolationInterface;
+use Drupal\Core\Field\Plugin\Field\FieldWidget\NumberWidget;
+
+/**
+ * Plugin implementation of the 'mode_number' widget.
+ *
+ * @FieldWidget(
+ *   id = "mode_number",
+ *   label = @Translation("Mode number field"),
+ *   field_types = {
+ *     "integer",
+ *     "decimal",
+ *     "float"
+ *   }
+ * )
+ */
+class ModeNumberWidget extends NumberWidget {
+
+  /**
+   * {@inheritdoc}
+   */
+  public static function defaultSettings() {
+    return array(
+      'placeholder' => NULL,
+      'min' => NULL,
+      'max' => NULL,
+      'default_value' => NULL,
+      'step' => NULL,
+      'prefix' => NULL,
+      'suffix' => NULL,
+    ) + parent::defaultSettings();
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function settingsForm(array $form, FormStateInterface $form_state) {
```

Comment:
Yeah after I commented I forgot the whole reason we need a widget is the core doesn't provide these settings.